src/tools/tc_utils/tc_rmw/tc_rmw_wind_converter.cc

```diff
@@ -0,0 +1,242 @@
+//   000   05/11/22   Albo           Pulled the wind conversion into a class
+//
+////////////////////////////////////////////////////////////////////////
+
+
+#include "tc_rmw_wind_converter.h"
+#include "series_data.h"
+#include "vx_regrid.h"
+
+////////////////////////////////////////////////////////////////////////
+
+static void wind_ne_to_ra(const TcrmwGrid&,
+			  const DataPlane&, const DataPlane&,
+			  const double*, const double*, double*, double*);
+
+////////////////////////////////////////////////////////////////////////
+
+void TCRMW_WindConverter::_free_winds_arrays(void)
+{
+  if (_windR != NULL) {
+    delete [] _windR;
+    _windR = NULL;
+  }
+  if (_windT != NULL) {
+    delete [] _windT;
+    _windT = NULL;
+  }
+}
+
+////////////////////////////////////////////////////////////////////////
+
+TCRMW_WindConverter::TCRMW_WindConverter(void) :
+  _windR(NULL),
+  _windT(NULL),
+  _foundUInInput(false),
+  _foundVInInput(false),
+  _unitsU("Unknown"),
+  _conf(NULL),
+  _computeWinds(false) {
+}
+
+////////////////////////////////////////////////////////////////////////
+
+TCRMW_WindConverter::~TCRMW_WindConverter(void) {
+  _free_winds_arrays();
+}
+
+////////////////////////////////////////////////////////////////////////
+
+void TCRMW_WindConverter::init(const TCRMWConfInfo *conf) {
+  _conf = conf;
+  _computeWinds = _conf->compute_tangential_and_radial_winds;
+  _free_winds_arrays();
+  _windR = new double[_conf->n_range*_conf->n_azimuth];
+  _windT = new double[_conf->n_range*_conf->n_azimuth];
+  _foundUInInput = false;
+  _foundVInInput = false;
+  _unitsU = "Unknown";
+  _uIndexMap.clear();
+  _vIndexMap.clear();
+
+  if (!_computeWinds) {
+    return;
+  }
+
+  VarInfo* data_info = (VarInfo*) 0;
+  for(int i_var = 0; i_var < _conf->get_n_data(); i_var++) {
+    data_info = _conf->data_info[i_var];
+    string varname = data_info->name_attr().string();
+    string varlevel = data_info->level_attr().string();
+    if (varname == _conf->u_wind_field_name.string()) {
+      _uIndexMap[varlevel] = i_var;
+    }
+    else if (varname == _conf->v_wind_field_name.string()) {
+      _vIndexMap[varlevel] = i_var;
+    }
+  }
+  // test for consistency
+  if (_uIndexMap.size() != _vIndexMap.size()) {
+    mlog << Error << " error uneven number of ugrid/vgrid, no wind conversion\n";
+    mlog << Error << _conf->u_wind_field_name.string() << " had " << _uIndexMap.size() << " inputs\n";
+    mlog << Error << _conf->v_wind_field_name.string() << " had " << _vIndexMap.size() << " inputs\n";
+    _computeWinds = false;
+  }
+  map<string,int>::const_iterator iu, iv;
+  for (iu=_uIndexMap.begin(), iv=_vIndexMap.begin(); iu!=_uIndexMap.end(); ++iu, ++iv) {
+    if (iu->first != iv->first) {
+      mlog << Error << "error ordering of ugrid/vgrid input levels not the same, not implemented, no wind conversions\n";
+      mlog << Error << "    "  << iu->first  << " " << iv->first << "\n";
+      _computeWinds = false;
```

Comment:
Everywhere else in MET, all ERROR messages are followed by a call to exit(1) to return bad status.
Recommend either exiting here, or change the "Error" messages to "Warning" instead.